Applications/Spire/Include/Spire/Ui/OrderTypeComboBox.hpp

```diff
@@ -16,12 +17,25 @@ namespace Spire {
       */
       using SelectedSignal = Signal<void (Nexus::OrderType type)>;
 
-      //! Constructs a OrderTypeComboBox.
+      //! Constructs an OrderTypeComboBox.
       /*!
         \param parent The parent widget.
       */
       explicit OrderTypeComboBox(QWidget* parent = nullptr);
 
+      //! Returns the currently selected OrderType.
+      Nexus::OrderType get_current_order_type() const;
+
+      //! Returns the currently activated OrderType, or the currently selected
+      //! OrderType if there is no currently activated OrderType.
+      Nexus::OrderType get_last_order_type() const;
```

Comment:
This method looks suspicious, can you go into some detail why you have it?

Reply:
The StaticDropDownMenu triggers a selection signal when it loses focus, and if there is an item that was activated (using up/down or charcter key press), that item gets submitted as the selected item. The KeyBindingItemDelegate closes the editor (combo box) when the tab key is pressed, so the editor doesn't confirm the currently activated item when it loses focus.

This method provides a way to determine the intent of the user when the editor is closed. Either they explicitly selected an item and the function will return the current item, or the function will return the last activated item when the editor is closed.

An alternative would be to have the get_current_item() behave like the get_last_item(), but there may be inconsistencies if get_current_item() is called while the user is currently selecting a value. In this case, get_current_item() would return the currently activated item even though the user hasn't confirmed it.

Another alternative, assuming the editor is closed before setModelData is called, would be to have the StaticDropDownMenu's last activated item committed on a closeEvent, if the menu has a cell style.

Reply:
Okay one issue is a matter of terminology, a ComboBox has a current and an active item.

When a ComboBox loses focus, the active item becomes the current item.

Reply:
That's true, but from a technical perspective when the editor is closed it doesn't receive a focus out event, so the active item doesn't get confirmed as the current item.

Without modifying the API, my second suggestion would probably be the best way to fix this. To prevent potential issues with both a focus out and close event triggering a double selected signal, the active item would become the current item on a close event only when the ComboBox has the cell style applied to it.

Reply:
Okay, when a ComboBox is closed, the active item becomes the current item as well. Will that help fix it?

Reply:
After testing, the delegate's setModelData is called before the call to the ComboBox's closeEvent.

Reply:
That's good news right?

Reply:
Oh shoot that's bad news isn't it?

Reply:
At any rate, feel free to think up an alternative solution and if you don't think there's a reasonable solution, then go with your proposal of having ```get_current_item``` behave like ```get_last_item```.

Reply:
I'm looking into other events to see if one of them can be used as a trigger to update the current item. Ideally I'd remove get_last_item() and just have get_current_item() as it currently is, but if I can't find a way to encapsulate this in the ComboBox then I'll update get_current_item() to work like get_last_item().

Reply:
Agood... ... agood...

Reply:
Looking into this, the editor is closed entirely within the signal/slot system and the events for the Qt event system aren't sent until after the data is committed.

I don't see a way to implement this aside from updating get_current_item() to use the last active item.

Reply:
Go for it.